pkg/host/internal/sriov/sriov.go

```diff
@@ -456,7 +458,18 @@ func (s *sriov) configSriovVFDevices(iface *sriovnetworkv1.Interface) error {
 					linkType = s.GetLinkType(iface.Name)
 				}
 				if strings.EqualFold(linkType, consts.LinkTypeIB) {
-					if err = s.SetVfGUID(addr, pfLink); err != nil {
+					guid := utils.GenerateRandomGUID()
```

Comment:
I think you should remove all guid-related logic from this function. 
I would prefer to have a new `InfinibandInterface` and inject is as a dependency during the initialization of the `sriov` struct. Here you should call `s.infiniband.ConfigureVFGUID(addr, pfLink)` WDYT?